### ISSUE TITLE
Option to disable the link preview

### ISSUE THREAD

**Issue by Akamaru** [telegramdesktop/tdesktop#687]
Pls add an option to disable link previews.
![2015-04-30_20-04-52](https://cloud.githubusercontent.com/assets/8063440/7419572/2927004a-ef76-11e4-887e-cfb20c813a1b.png)

**Comment by Brawl345** [comment]
Yeah, they're really big and I think that your PC connects to the services like youtube? Not really a good feature for a privacy messenger.

**Comment by ppparadox** [comment]
+1

**Comment by OliverWillemse** [comment]
I would also very much like this feature!

**Comment by yuergen** [comment]
I concur. The link preview is annoying.

**Comment by Sanva** [comment]
+1 —I'm also really annoyed by this... it's really disturbing to be writing with someone talking about some web application we are developing and having that preview every time we mention the app by its domain name.

**Comment by auchri** [comment]
@Brawl345 your pc is not connecting to the services, instead the telegram servers are connecting to the different websites

**Comment by yuergen** [comment]
@auchri if the URI includes tracking information, information will leak to the site's operator.

**Comment by halcion** [comment]
please, i cannot stress enough how much i dislike the link preview, please remove it.

**Comment by Aokromes** [comment]
How about a dropdown with the following options:

"Preview on message" (current mode)
"Preview on mouseover"
"Preview on click"
"Preview disabled"

**Comment by brantje** [comment]
+1, i expected an option to turn it off when they introduced link previews...
Also, as @Brawl345 mentioned, it also gets youtube / facebook links, not good for your privacy

**Comment by animalillo** [comment]
+1000

**Comment by Aokromes** [comment]
Maybe the setting can he also be done per group or contact overwriting general settings.

**Comment by tech-no-logical** [comment]
+1 this should really be optional

**Comment by ezdev128** [comment]
Please disable this [censored] preview feature!!!
Today I had an issue when trying to send auth link (to recover password to one of my online services) and I  was unable to do it because your built-in crawler already gets this link, therefore remote side think you have already checked link and doesn't allows to change password. My account has been blocked for fraud attempts after 3rd unsuccessful try.
This is awful.

@telegramdesktop how much signatures needs from unhappy persons to disable this [censored] feature? 10? 20? 100? 10000??? Don't you have an eyes or don't you read the conversation??? This question related to https://github.com/telegramdesktop/tdesktop/issues/807

**Comment by ralesk** [comment]
:+1: would love to have a setting to disable images sent/received and link previews

**Comment by ezdev128** [comment]
@ralesk that should be by default, I mean it should be optional.
Also it would be good to have some document types list to preview, e.g., allow preview only for *.html,*.png|gif|jpeg|..., ..., youtype url type (like /watch?v=hash) and other, but NOT ALL THE LINKS!!!

**Comment by mbravorus** [comment]
+1 on both off by default and per chat option to turn on

**Comment by OscarL** [comment]
+1

I have a slow internet connection, and that seems to prevent the link preview and "X to cancel" to show up in due time, so I end up getting previews when I DO NOT want them. Please implement this!

**Comment by HooFoo** [comment]
+1
 I suggest you solution to resolve this situation. 
1) Application settings needs new toggle - NSFW mode
2) If this toggled all pictures in all chats is under monochrome overlay with opacity value ~80%
3) When you hover you pointer overlay changes their opacity to 0% and you see original file.

**Comment by ralesk** [comment]
@HooFoo I don't think a hover revealer with any kind of default visibility is sufficiently safe for work.
Also, I suppose it's the client that looks up the contents of the remote URLs and what not — in this case it's doubly troubling that one ends up “browsing” websites that one does not have any control over.

The only right way to do SFW mode is to not do any lookups and not display any previews.

**Comment by HooFoo** [comment]
@ralesk I understand your position and I agree with it. But maybe this function (overlay) will be realised as "light SFW" mode? I am a supporter of this approach because you can control what content is safe, because you can see it.  I attached example: opacity of overlay is 85%. 
![default](https://cloud.githubusercontent.com/assets/3982887/10821273/9980ac58-7e62-11e5-9809-93a045818f07.png)

**Comment by poiuty** [comment]
+

**Comment by Dimtar** [comment]
How is this still an issue over 6 months later?

**Comment by Aokromes** [comment]
Use http://i.imgur.com/oSglMLl.png

**Comment by Dimtar** [comment]
Thanks Aokromes <https://github.com/Aokromes>

It's a start but it's not really quite there yet as there is still a
slightly blurred preview.

On Saturday, 16 January 2016, Aokromes <notifications@github.com> wrote:

> Use http://i.imgur.com/oSglMLl.png
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/telegramdesktop/tdesktop/issues/687#issuecomment-172170968>
> .
>


-- 
--
Thanks,
Daniel

**Comment by nthx** [comment]
+1000 on removing link preview
while keeping preview of images

**Comment by ptramsey** [comment]
+1.  I would use telegram 10x more if I felt that I got any say on what other people displayed on my device's screen.  The lack of this feature is why I don't feel I can log into telegram in public.

Other random people on the internet should not be able to display images on my monitor without my consent.

**Comment by fracarma** [comment]
+1

**Comment by youkoso** [comment]
Need it so much.

**Comment by ImNexus** [comment]
+100500

**Comment by auchri** [comment]
Please use the new [reactions of GitHub](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments) on **the first entry of this issue** instead of creating unnecessary `+1` comments.

I deleted all +1's

**Comment by tech-no-logical** [comment]
@auchri thereby seemingly decreasing the relative importance of this issue. although they did not contribure, the +1's did have meaning...

although I doubt this issue will ever be fixed. it's been over a year.

**Comment by auchri** [comment]
writing +1 or using the reactions makes no difference.

**Comment by stek29** [comment]
@auchri `+1`s notify everyone who is watching, but reactions won't

**Comment by ralesk** [comment]
All the better.  Less noise, more content (should there be actually any on this bug)

**Comment by kode54** [comment]
I don't know about anyone else in this topic, but I am now seeing imgur links previewed as extremely low resolution, horribly blurred images, both on Telegram Desktop and on the Android app, which I know is something else entirely. It's probably Imgur's fault.

**Comment by stalks** [comment]
What is the hold-up on this issue? Other less important issues are fixed quickly, but this one remains 1year+ later. Does this go against some philosophy from a higher-up at Telegram? What can we do to change their mind? If the answer is nothing, then why not close this ticket and say so.

I work in hosting, and a lot of messages have a domain in the text. This is getting old: http://i.imgur.com/K3JTHkk.png

**Comment by rlenders** [comment]
Every damn time I type ASP.NET I want this feature more and more.

**Comment by auchri** [comment]
@rlenders 

![image](https://cloud.githubusercontent.com/assets/5092164/15327173/d21bd09c-1c50-11e6-82af-6b4b5213a2a5.png)

:stuck_out_tongue_winking_eye:

**Comment by Zenexer** [comment]
@auchri, that's extremely inconvenient when you're talking about a domain/technology frequently, especially since it takes time for the preview of the preview to show up.  Sometimes it never appears at all.

As for privacy, Telegram never makes requests to any of these URLs, so there's no privacy issue.  It doesn't matter whether there's tracking or anything else in the URL.  Telegram's servers make the request and then forward the preview to all the clients.  The website has no way of knowing who saw it (or any other information, for that matter).  I've tested this quite thoroughly.

**Comment by ralesk** [comment]
> As for privacy, the Telegram client never makes requests to any of these URLs, so there's no privacy issue. It doesn't matter whether there's tracking or anything else in the URL. Telegram's servers make the request and then forward the preview to all the clients. The website has no way of knowing who saw it (or any other information, for that matter). I've tested this quite thoroughly.

Thanks for checking, that's good to know.

**Comment by johonaton** [comment]
This really should be a high priority...not allowing the disabling of chat media makes Telegram basically useless in a SFW setting as there's no opt-out for displaying media. Privacy aside with respect to the server requests, this is still a major problem.

**Comment by mrsofia** [comment]
Please create this feature. Telegram auto-clicks each sent link in order to gather some data and render the preview. I want to use Telegram to automatically send work alerts to my phone, but each alert has a link that, when clicked, tells the alert system that I've acknowledged the alert. When Telegram auto-scans the link to create a preview, it clicks the link thereby telling the alert system that I've acknowledged the alert. You can see the problem with this - I'm automatically acknowledging alerts before I've even looked at them. Enabling this feature in a given chat will solve this problem and enable more mass market adoption of Telegram for use cases like this. Thank you.

**Comment by auchri** [comment]
@mrsofia Bots are able to disable the preview: https://core.telegram.org/bots/api#sendmessage
So you should create a bot to send the messages.

**Comment by mrsofia** [comment]
@auchri Good to know that works, but hopefully this is a feature the team is planning on implementing. Creating a bot would require an email event-listener as the work alerts are sent through a carrier-specific email address that sends the email to my phone as an SMS. Right now I'm using a 3rd-party bot that provides an email address that sends the email to a Telegram group, which would provide the solution I need if I could turn off the link preview. Building my own bot would require a back-end email listener which I don't really have the time for. I'll probably implement this in Slack for now since they have those features. Thanks for the response.

**Comment by ThinkPadNL** [comment]
This really annoys me. I use Telegram (webclient) frequently, and the auto preview of URL's is making it harder for my chat partner to read the conversation in the layout as i meant it.
Please create an option to disable the link preview!

**Comment by lees** [comment]
+1

**Comment by stek29** [comment]
@ThinkPadNL please report the issue to [webclient repo](/zhukov/webogram) too then.

**Comment by ThinkPadNL** [comment]
@stek29 Thanks, i did that: https://github.com/zhukov/webogram/issues/1216

**Comment by foysalit** [comment]
Hey all, just submitted a PR on the web client in case anyone's interested in taking a look. https://github.com/zhukov/webogram/pull/1228

**Comment by dhowe** [comment]
Would be great to land this... it is by far the most annoying 'feature' of telegram's otherwise-excellent UI

**Comment by robotex82** [comment]
+1

**Comment by vdmit** [comment]
+1 for this feature.

**Comment by 6r0m** [comment]
It's so annoying - close this preview every time!!! because text become not good readable((

**Comment by dhowe** [comment]
why is no one responding here?

**Comment by 6r0m** [comment]
because Durov is a russian guy, need to write like this - "Пожалуста, реализуйте возможность скрывать автоматическую вставку превью ссылок"

**Comment by fgjnghj** [comment]
Будь ласка, реалізуйте можливість приховувати автоматичну вставку превью посилань.

**Comment by stek29** [comment]
@dhowe why should someone?

**Comment by stalks** [comment]
Scroll up to the first post and click the thumbs up. It'll then rate higher in the (list of other enhancements)[https://github.com/telegramdesktop/tdesktop/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc]

**Comment by stek29** [comment]
@stalks `label:enhancement` :)

**Comment by joy4eg** [comment]
+1

**Comment by brianleejackson** [comment]
I can't tell you how many clicks I waste on this per day... The thing is, this needs to be on a **per channel basis**. My entire team wants this, so I know there are others out there.

**Comment by m-legrand** [comment]
The Telegram's system of bot is so convenient for team of programmers who:

 - use a lot a hyperlinks;
 - really don't want to bother clicking on a cross for each and every message they write;
 - would love to avoid scrolling for nothing when watching what has been said since their last passage.

How can an absence of option be so difficult to patch in? At the very least put a cross even once the message has been sent so one can clean up the mess afterwards.

**Comment by vicnbt** [comment]
I've removed it with adblock in web version, 
may be this simple solution will help someone

**Comment by stek29** [comment]
@vicnbt it may help people -- can you share it at github.com/zhukov/webogram?

**Comment by auchri** [comment]
See #3330

**Comment by vdmit** [comment]
Collapse feature is not what personally I expected. I just wanted to have links just as simple blue links, nothing more, and, moreover, don't do any http requests for these links. So I don't think this issue could be closed.

**Comment by kirsan31** [comment]
@vdmit I agree.

**Comment by auchri** [comment]
> moreover, don't do any http requests for these links

The requests are done by the server.

**Comment by shrpne** [comment]
In 3330 links still cluttered, I prefer option to fully disable rich links

**Comment by animalillo** [comment]
I agree with this

El 27 de abril de 2017 17:10:02 CEST, shrpne <notifications@github.com> escribió:
>In 3330 links still cluttered, I prefer option to fully disable rich
>links 
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/telegramdesktop/tdesktop/issues/687#issuecomment-297743661

-- 
Enviado desde mi dispositivo Android con K-9 Mail. Por favor, disculpa mi brevedad.

**Comment by stek29** [comment]
#3330 just allows to unhide hidden previews in case you want to see them. So there's an option to disable or enable link previews by default, and to hide/unhide specific ones.

**Comment by Akamaru** [comment]
3330 is not what I (and many other) want. So why is this closed? Please reopen it...

**Comment by dhowe** [comment]
Agree with @Akamaru

**Comment by stek29** [comment]
@Akamaru the only difference is that #3330 would provide a way to unhide specific preview, while this one would force you to enable them in settings globally. Or are there any other differences I haven't thought of?

**Comment by vdmit** [comment]
@stek29: I could only repeat my previous comment https://github.com/telegramdesktop/tdesktop/issues/687#issuecomment-297739648:
we all **want to see simple blue links and don't waste space on the screen by preview box**. Is it too hard to realize? That's why collapsed form is not aceptable.

**Comment by kirsan31** [comment]
@stek29 No.
![image](https://cloud.githubusercontent.com/assets/17767561/26539771/d6d8e83e-4456-11e7-92d9-2414dcc2985c.png)
We don't need red, we only need green :-)

**Comment by stek29** [comment]
Why no one in this thread thinks about actual implementation? You just keep saying "I want this" without even thinking of how it can be implemented and what it would affect.
Both #3330 and this request same thing -- an ability to hide/collapse/disable web previews. 
The only difference is some people want it enabled by default, and others want it disabled by default. Just one Boolean in settings. 
That's why it's impractical to fix one without fixing another.

**Comment by kirsan31** [comment]
@stek29 We want ability to disable some current functional - link preview - one Boolean in settings ;) But #3300 want current functional improvements - ability to collapse link preview. Imho, our request more simple to implement, and it can easy be done first. Then, if still needed, authors can implement collapse individual preview feature... 
But if collapse feature is in priority, than yes, while implementing it, it's good idea to implement our request to.

**Comment by vdmit** [comment]
@stek29: Some time ago, I've even found place in code when this option should be checked. But, I failed to build tdesktop on Ubuntu 16.04 regardless of all my SE experience :( I've wasted several hours, but got finally stuck on 7-th or 8-th step of this "short, simple build instruction" https://github.com/telegramdesktop/tdesktop/blob/dev/docs/building-cmake.md Of course, I need another try

**Comment by ralesk** [comment]
Also, if anything, the new bug does not invalidate this one, nor does it make this one "done".  Most of the discussion about the issue happened here, closing this bug (instead of the newer one) is simply sweeping it under the rug.

But of course the other bug is also not in any serious consideration as per last comment in April, so it's not like it matters terribly much either way.

**Comment by maxim-ge** [comment]
+1, please remove this or give an option to disable!

**Comment by grasmanek94** [comment]
Why is this closed? I vote for reopen. This is a real issue. Link previews are generated and users can't control it!

**Comment by stek29** [comment]
@grasmanek94 If you're the sender, you can disable preview by closing the suggestion. If you're reciever, see #3330.

**Comment by grasmanek94** [comment]
@stk29 What is people want only the text and absolutely no preview, even not a title and short description? I cannot disable that anywhere in the options. I'm going to try and make this in my own branch, because I reaaaaallllyy want that and I have the feeling it's not going to be added anytime soon.. :(

**Comment by stek29** [comment]
> requires user interaction in the chat itself 

No, #3330 doesn't require previews to be enable by default
Also, what's so wrong with preview? Why don't you like it?

**Comment by lees** [comment]
I don't like it, because I don't need it. If I want to look up the link, I'll do it by myself. And #3330 not solving my problem. I vote for reopen. And I really don't understand why it is so hard to understand. You need a scenario when I don't need a previews?

**Comment by grasmanek94** [comment]
> Also, what's so wrong with preview? Why don't you like it?

Well when I use Telegram Desktop on my desktop, I click the links my friends send me anyway, so why would I want to waste space and have less compact chats? It feels good when I can access as much information as I want without scrolling. And this is especially true when running a multi monitor setup while I'm gaming, if I need information and I need to scroll, I'm f****d, especially in first person shooters.

**Comment by grasmanek94** [comment]
If someone wants a version of Telegram with all link previews disabled you can find it [here](https://github.com/grasmanek94/tdesktop/tree/nopreview), I'm not going to make a pull request because I don't know if this is the correct solution to the problem.

**Comment by vdmit** [comment]
@grasmanek94, you've done a great work. However, I've studied your patch and it appeared a bit complicated to me: actually we don't need to fix (hide) preview itself and change it's geometry. After some code exploring, i've done this patch:
https://gist.github.com/vdmit/f490be029544d728657ac6f2e53fb61c
I've not tried to build your fork yet, but it seems that it has more-or-less same effect.
With my dirty hack, it shows _simple blue links_ for web pages, and small thumbs for images (which is acceptable).

P.S. Build sequence is terrible, but finally, I've defeated it!

**Comment by grasmanek94** [comment]
@vdmit ah good job! I was actually looking for such code, I didn't know to look in history_message for that. So I just did it by trial and error :) After taking around 3 hours to setup the build environment for VS2015 I really didn't have much time left to study the Telegram code thoroughly. Thanks for the much better solution, works exactly (visually) as my solution. We should make a boolean in settings which controls the break behaviour but I don't think I can free up some time to figure out how to do that. So lets wait it out :p

**Comment by grasmanek94** [comment]
@vdmit I have compiled your changes into a Windows Release executable [here](https://github.com/grasmanek94/tdesktop/releases/tag/v1.1.7), if anyone can't compile the source for themselves and really wants a binary with @vdmit 's patch, there you go.

**Comment by dhowe** [comment]
How about an osx version?

**Comment by vdmit** [comment]
@dhowe currently I don't have osx environment to build it. Maybe other users can help.
However, the build instructions seem to be correct, just follow this: https://github.com/telegramdesktop/tdesktop/blob/dev/docs/building-xcode.md
In fact, they are not complicated, but there are a LOTS of steps and no complete build script supplied.
(I'm going to make such all-in-one build script for Linux and push it upstream)
So, read build messages carefully and don't skip any errors: if something went wrong, that means that you had mistaken in libraries layout above, etc. Finally, allocate couple of hours, brew a large cup of coffee, and... go on! =) It's a great sense when you made something good for the Community! :)

**Comment by stek29** [comment]
@vdmit @dhowe https://github.com/stek29/homebrew-tdesktop
But I think produced app is dynamically linked to some libs, I might have to look into that

**Comment by vdmit** [comment]
Good news, now I have some progress with settings:
![no-web-preview](https://user-images.githubusercontent.com/1774752/27254945-3724efb8-539c-11e7-99d4-650734d7445c.png)
Surprisingly, it was quite easy to extend settings box. There are some problems with layout left... well, copypaste is your friend, but not the best one :) I'm pretty sure that I'll fix this.

When I started new (own) version, it appeared to work like fresh installation (with sms passcode and default settings). I'm not sure if it's correct, but... still, it works.

**Comment by stek29** [comment]
@vdmit Nobody said adding settings is hard -- it's super easy _(but having to deal with manual layout is kinda gross...)_
The problem is that settings adding more and more settings is not a sane thing to do -- not for Telegram. Telegram is not WeeChat or Pidgin.

Also, if I understood correctly, you're just setting height of preview to 0, or am I wrong?

**Comment by vdmit** [comment]
@stek29 
1. On settings number:
Yeah, you're right. Good app should have no control panel like aircrafts do.
But how many times a day you visit your settings? Typically it's zero, I suppose. And how many times do you deal with links? Everyday. In our work, we send tons of links among our collegues, and these previews waste all the history area, and looking through the history makes the mousewheel burn.

2. On implementation:
No, I don't deal with geometry at all. I just don't load preview itself - just like in case of empty page. I've almost finished my fix with 2 new checkboxes and it looks pretty cool, so I'm going to make a PR in a while. You may observe implementation there.

**Comment by shrpne** [comment]
@stek29 one more checkbox on the settings page is bad, but one more toggle button on every link is pretty fine, right?

**Comment by grasmanek94** [comment]
@stek29 going by that logic, why does Telegram fetch web content if it's not a web browser? It's fine if Telegram has a well defined boundary as to what it should do and what not, but wouldn't it be better to expand those boundaries to allow for useful and significant features? In my opinion yes.

**Comment by stek29** [comment]
@shrpne Why not?

@vdmit 
> No, I don't deal with geometry at all. I just don't load preview itself - just like in case of empty page.

Well, TDesktop loads it tbh. You just don't process it I guess :)

**Comment by Dimtar** [comment]
@grasmanek94 I mean this with all respect but could you please explain how your build would be safe? I assume that it uses the telegram API as the desktop version does so credentials couldn't be taken?

**Comment by stek29** [comment]
@grasmanek94 just noticed your edited comment.
Please, have look at https://core.telegram.org/tsi/feature_philosophy. Tbh I highly doubt this would be merged.

> why does Telegram fetch web content if it's not a web browser

Telegram app does not fetch web content. It just shows whatever server provides to users.
And Telegram as a whole solution does fetch web content since it's useful. There are many cases: starting from not having to open the browser to look at something, and ending with search improvement.

Also, I don't think your example about 4K games fits here, but I'm not going to comment on that since it's not up for me or you to decide what features would be supported and what features won't.

**Comment by grasmanek94** [comment]
@Dimtar someone else would need to verify this by either compiling the patch and comparing the diffs (none if all settings are the same) or doing an assembly comparison.

@stek29 thanks for the link to the feature philosophy, very interesting and eye opening read. As for settings are evil, it's very true, but how could I determine if at least 5% of the users would use "disable link preview" option? I think there would be at least 5% but I have absolutely nothing to support that.

**Comment by stek29** [comment]
@grasmanek94 I don't know :(
We can't be sure if everyone who wants this feature had reached this GH repo.
It was requested to Support already more than a year ago, and it's still in TG's internal requested features list.

**Comment by vdmit** [comment]
It's in the top 10 by comments, however, and could be googled easily (that's why I'm here).
And even #277 (which irritates me too!) has currently "only" 57 likes for the first post, comparing to almost 90 likes here.

**Comment by stek29** [comment]
@vdmit I think he's a bit busy now with new features, and he would have a look at it after upcoming release.

**Comment by andrew-phi** [comment]
Please disable those damn annoying previews! Not only they steal privacy, but also valuable traffic! Add an OPTION, because USER should decide what HE needs!

**Comment by ulrickno94** [comment]
+1

**Comment by stek29** [comment]
@andrew-phi 
> they steal privacy

No, they don't. They are NOT fetched BY APP, BUT they're fetched BY SERVER.

> valuable traffic

MTProto is good at traffic usage. Really good. And those previews are usually sent in message, they're not fetched separately. To disable them you'd have to do that on server side, you can't just load part of server answer.

> USER should decide what HE needs

No, asking a user what (s)he wants isn't always a good thing. Neither is polluting settings with OPTIONs.

**Comment by shrpne** [comment]
@stek29 
> No, asking a user what (s)he wants isn't always a good thing. Neither is polluting settings with OPTIONs.

Not asking isn't always good too.

I like, that telegram team trying to make app clean and simple ass possible. 
But here is the problem, that link previews looks not clean at all, they are generated automatically and make chat window to look like a mess.

Adding toggle buttons to each link is not very useful.
If it will be opened by default, user will be need to close every preview in incoming and outcoming, it consumes too much time to keep chat clean.
If it will be closed by default, user will be need to click each time, when he want to see preview. If he needs to make a click, in most cases it will be faster to click straight on the link instead of preview button.

So I think this issue should be reopened and have more priority than #3330

**Comment by Gubarev** [comment]
I am using Telegram about a year and a half. I use it both for personal chats and for my work, so I chose Telegram as default messenger for my company. I see that Telegram gets better (super groups, voice calls, etc), however automatic link preview is the most annoying and biggest issue I ever experienced with Telegram. I would exchange all new features appeared past year just to disable preview. I think that disabling link-preview is the hottest and most desired feature and should be top-1 priority for Telegram dev team (after security updates of course).

Beside the fact it annoys me (I can handle my anger), it makes difficult to read messages for my interlocutors. In most cases people ignore the text of sibling messages if it contains link, because they are focused on title and image of preview. It leads to persistent misunderstanding. People simply do not notice messages right before or right after previews.

So I see only three options that may satisfy our needs:

1. Completely remove preview (ideal case).
2. Make it disabled by default. Users that want to see preview should enable preview for certain message (e.g. by clicking toggle button or display popup by mouse hover) or enable preview by default in settings.
3. Make possible for sender to choose to insert or not insert preview for his message.

Please, let us hope that it will be resolved once! I am afraid that despite other good features of Telegram we are really have to migrate to other message services.

**Comment by dhowe** [comment]
I've already moved back to whatsapp and signal b/c of this issues. What a shame...

**Comment by john-preston** [comment]
@Gubarev 3. is possible right now.

**Comment by cryptobug** [comment]
I need an option to remove link preview in Telegram completely. In settings, in config, in registry, whatever. It's a shame such a software team adds an option I don't need without disabling possibility. Looks like they say for a reason that if you don't buy product means YOU are the product. 
Will start using Telegram again only after this option is in place. Thank you.

**Comment by dariox2** [comment]
Do you know what a "one-time link" is?

**Comment by vuluc88** [comment]
pls make it SFW

**Comment by qadirsuh** [comment]
@vdmit where can I find the option you have sent screenshot above. 
I could't find it in Telegram Mac (Desktop) App.

**Comment by kode54** [comment]
Telegram Desktop (from desktop.telegram.org, or Mac App Store) is not equal to Telegram for Mac (from macos.telegram.org, or also the Mac App Store). One is a multi-platform app released for Windows, Mac, and Linux, and the other is written in Swift specifically for macOS.

You'll probably want to dupe this issue here if you want traction there:

https://github.com/overtake/TelegramSwift

**Comment by patoshii** [comment]
still no ability to disable url previews? kinda anonying. especially when NSFW images are rendered.

**Comment by Bacco** [comment]
It seems to me that the link preview is generated on the server, but requested on the client. A simple checkbox in settings appears to be the more sensate way to resolve the issue.

Clicking on the X can be enough to some scenarios, but in a conversation with heavy link exchange, it can become annoying quickly.

Also, recently I was testing an API with help from a friend, and needed to switch to another chat, because the links would be compromised by Telegram server, that would consume the API nonce before even the link reach the other party. The [x] wouldn't resolve the issue, because the damage already has been done.

I hope @auchri and other devs think a lot more in this scenarios and help us with a solution.

Regards
Bacco

**Comment by qadirsuh** [comment]
BTW. if you are sending the link programmatically then its possible to remove/hide/disable link preview. as follows

`disable_web_page_preview: true`

javascript function to call sendMessage API.

```
function sendMessage(chatId, message, form) {

    var formMarkdown = {
        parse_mode: "Markdown",
        disable_web_page_preview: true
    };

    if (!form) {
        form = formMarkdown;
    } else {
        form.parse_mode = "Markdown";
        form.disable_web_page_preview = true;
    }

    telegramBot.sendMessage(chatId, message, form).then(function (response) {
        console.log("response", response);
    }).catch(function (err) {
        console.log("sendMessage err:", err);
    })
}
```

**Comment by stek29** [comment]
@qadirsuh that's bot api, btw.

**Comment by vsvasya** [comment]
Why is it closed?
Please, fix it! Fighting with previews in every message I sent is very annoying.

**Comment by patoshii** [comment]
just make more of the same topic. if you want to get their attention.

**Comment by ItsOnlyMeNL** [comment]
Since 30 Apr 2015 this exists.. still not added. Add this option for the user to disable all link previews for send and receive.

**Comment by spocko** [comment]
I don't understand why this has been ignored and closed. It's obviously an option that a lot of people want. There are 122 upvotes on the first post, and those are just people who are technically inclined enough to use github. This really should be reopened and addressed.

**Comment by dhowe** [comment]
A bit offensive to all the people who took the time to comment here (anyone else for signal?)

**Comment by andrew-phi** [comment]
I think tg owners just don't give a duck about what people really want. They just make what THEY want. :-|

**Comment by nsa** [comment]
It is sad to see that this community(there are 51 people participating in this issue) is being ignored.

**Comment by Bacco** [comment]
For most people this is only a nuisance, but there are situations where we simply can't use Telegram to store some links (like password reset ones, confirmation of some operation), because the previewer will trigger the action even if the user doesn't want to click it yet.

**Comment by suborg** [comment]
Please make it possible to disable link preview. It can even destroy links that work by count.

**Comment by OhSoGood** [comment]
+1. It completely pollutes the display when you exchange several links.

**Comment by Vondelphia** [comment]
+1

p.s. - i [wonder if unigram has this option](https://github.com/UnigramDev/Unigram), it probably does....

**Comment by jamsmd** [comment]
+1

**Comment by tazotodua** [comment]
**its now available !**
use `disable_web_page_preview`  parameter:
https://core.telegram.org/bots/api#editmessagetext

**Comment by RememberTheAir** [comment]
@tazotodua that has always been there. It's jus a flag clients use to send a message without the webpage preview. You can't change other people's messages using the API

**Comment by NikEyX** [comment]
How can it be 4 years and there is still no option for that? 

In discord you surround a link with <> brackets in order to make it non-expandable. Why is this not done for Telegram yet? cmon!

**Comment by dhowe** [comment]
these guys don't care - just switch to signal

**Comment by ttodua** [comment]
@RememberTheAir ah, i misunderstood that. thanks (btw, please in your post, correct my username, it should be "ttodua"). thanks !

**Comment by spocko** [comment]
I am quite baffled that this simple request has not been addressed 5 year later. The justification for ignoring it seems to a general philosophy that "settings are evil". That philosophy cannot be blindly applied as an excuse to ignore all feature requests that require settings. New settings should be provided selectively as needed, and this is a case where there is a clear need

As others have already noted earlier in this thread, this is a functional issue, not just a UI preference. One-time-use links simply cannot be sent via telegram because the server accesses the link.  #3330 is not a solution to this problem, because it doesn't stop the link from being accessed.

There is also a UI aspect to this issue. Some users are not concerned about the link being accessed, they just don't want to see previews in their chats. That is also a valid user request, even though it is a preference. In this case, something like #3330 could help, but that approach also adds complexity in the chat UI. Users look at the chat UI all the time, and rarely look at the settings. So if the goal is to reduce overall complexity, then it seems better to address this in settings rather than having it in every single message that contains a link preview.

Lastly, there are comments earlier in this thread saying that link previews do not compromise privacy/security because they are generated by the server and not the client. I'm not sure that's true in all cases. If a link is a tracking link that uses a unique URL, then any access to that link is associated with the original recipient for whom the URL was created. Of course it's not wise to send such links if you don't want to be identified, but in practice, such a link could get copied unintentionally, or by a user who doesn't know what a tracking link is.

According to the philosophy page linked earlier ( https://tsf.telegram.org/manuals/feature_philosophy ), Telegram's top two attributes are speed and security. Reducing complexity is also supposed to be a high value.  Automatically inserting link previews into messages impairs all three of those objectives in my opinion.

Please consider reopening this issue.

@auchri @stek29

**Comment by serjepatoff** [comment]
Please turn this BS feature off by default. It eats up precious screen space and hurts privacy.

**Comment by danlugo92** [comment]
Please offer an option to disable link preview.

**Comment by Stealer0** [comment]
I want to always send links without previews. I don't use emoji, stickers and keep our chats pretty organized with text-only. How long do I have to keep editing messages and click X on each link to not look like an ignorant?

**Comment by siran** [comment]
why was this request closed?

**Comment by siran** [comment]
TLDR click on the little x after pasting the link
this was already addressed by the Telegram team: https://telegram.org/blog/link-preview
also, in most cases I don't think this is a security/privacy issue and the link is scanned anonymously by Telegram's servers

**Comment by kode54** [comment]
And the problem is, links that don't require confirmation to acknowledge their limited accessibility. For instance, a link that can only be clicked once, and it acknowledges Telegram's scanner as that one click.

**Comment by retiolus** [comment]
Experts Warn of Privacy Risks Caused by Link Previews in Messaging Apps
https://thehackernews.com/2020/10/mobile-messaging-apps.html

**Comment by testerik** [comment]
I don't know if anyone has already requested this feature: Please provide an option to disable the link preview.

(Telegram Desktop, Linux)

**Comment by benkinooby** [comment]
Please provide an option to en-/disable link preview by default for links. The preview rarely adds anything of value, clutters the chat and is a privacy risk.

**Comment by l0rb** [comment]
I want to always send links without previews. How long do I have to keep waiting for the preview to load and clicking X on each link?

**Comment by mwuttke97** [comment]
What the heck?! This has been ignored since 2015?!

**Comment by dsvyrjm** [comment]
Is that so hard? I'm tired to manually remove link preview every time

**Comment by benkinooby** [comment]
Hi @auchri 

I hope you don't mind me mentioning your name to get your attention. This issue was set "closed" by you in April 2017 and was not attended to since then. Please reopen this ticket, since it's still gaining interest, comments and reactions.

Kind regards & Thank you in advance.

**Comment by RememberTheAir** [comment]
This is the intended behavior and it's unlikely to change because of the reactions to this github issue. I suggest to open a ticket on https://bugs.telegram.org about this request. As long as the discussion stays here and as long as other Telegram apps don't offer the same option or adopt a different behavior, there's little hope to see such a feature being implemented on tdesktop. Although we recently had some improvements for link previews that other clients still don't support

**Comment by benkinooby** [comment]
For reference a related issues:
Title: The link preview appears even if it was disabled
https://bugs.telegram.org/c/2528

> As long as the discussion stays here and as long as other Telegram apps don't offer the same option or adopt a different behavior, there's little hope to see such a feature being implemented on tdesktop.

This is valuable input, because it motivates the team's decision to close this bug. That may not make it (i.e. auto-downloading of a link preview) acceptable, but at least understandable and may have prevented some of the less productive, rather emotional comments on this issue, which where certainly sparked in part by a lack of communication.

Hopefully, your comment will lead to an improvement of the issue at hand and the communication surrounding it.

**Comment by OscarL** [comment]
The more appropiate issue in "bugs.telegram.org" that I could find was this one:

https://bugs.telegram.org/c/3195

I've added a comment there, linking back to this issue.

Let us hope some day we can have that darn opt out checkbox!
